sys/linux/bpf.txt

```diff
@@ -90,12 +91,15 @@ bpf_map_update_arg {
 bpf_map_delete_arg {
 	map	fd_bpf_map
 	key	ptr64[in, array[int8]]
+	value	const[0, int64]
+	flags	const[0, int64]
```

Comment:
@dvyukov Do these sort of zero-filled fields as the end of structures matter to syzkaller? If they do, we may need to add a much larger padding to all these anonymous structures because [`CHECK_ATTR()`](https://elixir.bootlin.com/linux/v5.3/source/kernel/bpf/syscall.c#L737) in the kernel checks the structure is filled with zeros in all unused fields and [`union bpf_attr`](https://elixir.bootlin.com/linux/v5.3/source/include/uapi/linux/bpf.h#L361) is actually quite large.

Reply:
Well, it will write 0s there. They also affect len/bytesize.
But I had an impression that CHECK_ATTR() only checks up to a particular field, no?

Reply:
E.g. the one you pointed to only checks up to:
https://elixir.bootlin.com/linux/v5.3/source/kernel/bpf/syscall.c#L724

Reply:
As far as I understand, it checks *from* a particular field, *up to* the end of `union bpf_attr`. In the case of `map_lookup_elem`, it checks that everything *after* `(void *) &attr->flags + sizeof(attr->flags)` is zero. Unless I missed something?

Reply:
OK, I think the way copying from user-space works:
https://elixir.bootlin.com/linux/v5.3/source/kernel/bpf/syscall.c#L2822
it auto-zeros the unused part (based on the size that we pass). Otherwise it would not be possible to pass the attr from user-space since it's constantly extended, so one doesn't know how much to zero.

Reply:
Such interface would not be backwards-compatible.
So hopefully we are fine?

Reply:
Yep, we're fine. There's just some additional overhead from copying these zero fields in kernel space, but I don't know if that can grow to something significant for syzkaller...

Reply:
That won't affect overall performance in any significant way.